src/test/kotlin/uk/gov/justice/digital/hmpps/courthearingeventreceiver/controller/EventControllerIntTest.kt

```diff
@@ -78,6 +101,43 @@ class EventControllerIntTest : IntegrationTestBase() {
       verify(telemetryService).trackEvent(TelemetryEventType.COURT_HEARING_UPDATE_EVENT_RECEIVED, expectedMap)
     }
 
+    @Test
+    fun whenLargeMessagePostToEventEndpointWithRequiredRole_thenReturn200NoContent_andPushToTopic() {
+      val str = File("src/test/resources/json/large-hearing-update.json").readText(Charsets.UTF_8)
+      hearingEvent = objectMapper.readValue(str, HearingEvent::class.java)
+
+      postEvent(
+        hearingEvent,
+        jwtHelper.createJwt("common-platform-events", roles = listOf("ROLE_COURT_HEARING_EVENT_WRITE")),
+      )
+        .exchange()
+        .expectStatus().isOk
+
+      val messages = courtCasesQueue?.sqsClient?.receiveMessage(ReceiveMessageRequest.builder().queueUrl(courtCasesQueue?.queueUrl!!).build())!!.get()
+      assertThat(messages.messages().size).isEqualTo(1)
+      val message: SQSMessage = objectMapper.readValue(messages.messages()[0].body(), SQSMessage::class.java)
+
+      val s3Reference: java.util.ArrayList<*> = objectMapper.readValue(message.message, ArrayList::class.java)
+      val mapper = ObjectMapper()
+      val s3Pointer = mapper.readValue(mapper.writeValueAsString(s3Reference.get(1)), LinkedHashMap::class.java)
+      assertThat(s3Pointer.get("s3BucketName")).isEqualTo(largeCasesBucketName)
+      assertThat(s3Pointer.keys).contains("s3Key")
+      assertThat(message.messageAttributes.eventType.type).isEqualTo("String")
+      assertThat(message.messageAttributes.eventType.value).isEqualTo("commonplatform.large.case.received")
+      assertThat(message.messageAttributes.messageType.type).isEqualTo("String")
```

Comment:
Would be great to see this test download the message from S3 and check the body.
I know it's already doing a lot!